pkg/v1/tkg/test/tkgctl/shared/package.go

```diff
@@ -0,0 +1,145 @@
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/log"
+)
+
+const (
+	getResourceTimeout  = time.Minute * 1
+	waitForReadyTimeout = time.Minute * 20
+	pollingInterval     = time.Second * 30
+)
+
+// create cluster client from kubeconfig
+func createClientFromKubeconfig(exportFile string, scheme *runtime.Scheme) (client.Client, error) {
+	config, err := clientcmd.LoadFromFile(exportFile)
+	Expect(err).ToNot(HaveOccurred(), "Failed to load cluster Kubeconfig file from %q", exportFile)
+
+	rawConfig, err := clientcmd.Write(*config)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create raw config ")
+
+	restConfig, err := clientcmd.RESTConfigFromKubeConfig(rawConfig)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create rest config ")
+
+	client, err := client.New(restConfig, client.Options{Scheme: scheme})
+	Expect(err).ToNot(HaveOccurred(), "Failed to create a cluster client")
+
+	return client, nil
+}
+
+func checkClusterCBS(ctx context.Context, mccl, wccl client.Client, mcClusterName, wcClusterName, infrastructureName string) error {
+	log.Infof("Verify addons on workload cluster %s", wcClusterName)
+	var err error
+
+	// Get ClusterBootstrap and return error if not found
+	clusterBootstrap := getClusterBootstrap(ctx, mccl, constants.TkgNamespace, mcClusterName)
+
+	// get package details from clusterBootstrap
+	cniPkgShortName, cniPkgName, cniPkgVersion, err := getPackageDetailsFromCBS(clusterBootstrap.Spec.CNI.RefName)
+	Expect(err).NotTo(HaveOccurred())
+	kappPkgShortName, kappPkgName, kappPkgVersion, err := getPackageDetailsFromCBS(clusterBootstrap.Spec.Kapp.RefName)
+	Expect(err).NotTo(HaveOccurred())
+
+	// verify clusterBootstrap details matches package install on workload cluster
+	verifyPackageInstall(ctx, wccl, wcClusterName, cniPkgShortName, cniPkgName, cniPkgVersion)
+	verifyPackageInstall(ctx, wccl, wcClusterName, kappPkgShortName, kappPkgName, kappPkgVersion)
+
+	// csi and cpi packages only appears in vsphere env
+	if infrastructureName == "vsphere" {
+		csiPkgShortName, csiPkgName, csiPkgVersion, err := getPackageDetailsFromCBS(clusterBootstrap.Spec.CSI.RefName)
+		Expect(err).NotTo(HaveOccurred())
+		cpiPkgShortName, cpiPkgName, cpiPkgVersion, err := getPackageDetailsFromCBS(clusterBootstrap.Spec.CPI.RefName)
+		Expect(err).NotTo(HaveOccurred())
+		verifyPackageInstall(ctx, wccl, wcClusterName, csiPkgShortName, csiPkgName, csiPkgVersion)
+		verifyPackageInstall(ctx, wccl, wcClusterName, cpiPkgShortName, cpiPkgName, cpiPkgVersion)
+	}
+
+	// loop over additional packages list in clusterBootstrap spec to check package information
+	if clusterBootstrap.Spec.AdditionalPackages != nil {
+		// validate additional packages
+		for _, additionalPkg := range clusterBootstrap.Spec.AdditionalPackages {
+			pkgShortName, pkgName, pkgVersion, err := getPackageDetailsFromCBS(additionalPkg.RefName)
+			Expect(err).NotTo(HaveOccurred())
+			verifyPackageInstall(ctx, wccl, wcClusterName, pkgShortName, pkgName, pkgVersion)
```

Comment:
"tkg-storageclass" package install is currently failing. Lets skip checking tkg-storageclass for now. Please add a TODO comment to remove this once the issue is fixed